### PR TITLE
Supplementary code based on the current metadata v14 code

### PR DESCRIPTION
**Pull request by Gkirito**
The `Type` field is missing in the MetadataV14 structure

### CODE REVIEW

**Comment by Gkirito** [comment]
I found `UCompact` struct already in the `types` folder.
According to [polkadot-js/api](https://github.com/polkadot-js/api/blob/master/packages/types/src/interfaces/scaleInfo/types.ts#L130)

``` ts
/** @name Si1LookupTypeId */
export interface Si1LookupTypeId extends Compact<u32> {}
```
The type `Si1LookupTypeId ` similar to type `Ucompact`.

File: types/metadataV14.go [hunk 1]
```diff
@@ -15,6 +15,7 @@ type MetadataV14 struct {
 	Lookup    PortableRegistry
 	Pallets   []PalletMetadataV14
 	Extrinsic ExtrinsicV14
+	Type      Si1LookupTypeID
```
Comment (NunoAlexandre) [inline types/metadataV14.go:18]: Where do you see this type? I am following the [latest v14 spec](https://github.com/polkadot-js/api/blob/48ef04b8ca21dc4bd06442775d9b7585c75d1253/packages/types/src/interfaces/metadata/v14.ts#L30-L34) and it's not there 🕵️

**Review by NunoAlexandre — COMMENTED** [review]
Thank you so much for taking the time, that's awesome!

A few things:

1. I don't see the field `Type` in the v14 for `MetadataV14`
2. Good call regarding `Si1LookupTypeID` extending `UCompact` 🎯 
3. Let's not leave commented code and delete it instead 🧹

File: types/metadataV14_portable.go [hunk 2]
```diff
@@ -77,19 +77,21 @@ func (d *Si0TypeDefPrimitive) Decode(decoder scale.Decoder) error {
 
 //------------------v1-----------
 
-type Si1LookupTypeID big.Int
+type Si1LookupTypeID struct {
+	UCompact
```
Comment (NunoAlexandre) [inline types/metadataV14_portable.go:81]: I realized that we already have `type Si0LookupTypeID UCompact`, ie., a type alias. If you don't feel strong about defining it as a struct, we would benefit from being consistent and have `type Si1LookupTypeID UCompact` instead.

File: types/metadataV14.go (hunk 1 above)
Comment (Gkirito) [reply types/metadataV14.go:18]: check master branch [master](https://github.com/polkadot-js/api/blob/0d0a230c860dfd9c2079dc341b0fa61387ad4b1d/packages/types/src/interfaces/metadata/v14.ts#L36-L40)

File: types/metadataV14_portable.go (hunk 2 above)
Comment (Gkirito) [reply types/metadataV14_portable.go:81]: Ok, But the inheritance of go language is a bit strange, If you use `type Si1LookupTypeID UCompact`, you have to rewrite the `encode` and `decode` methods belonging to `Si0LookupTypeID `. I use the structure method to make `Si0LookupTypeID ` inherit the methods of `UCompact`. 
Or you have other better ideas ?🤔

Comment (NunoAlexandre) [reply types/metadataV14_portable.go:81]: I am new to Go, didn't know that. Sounds good 👍

Comment (Gkirito) [reply types/metadataV14_portable.go:81]: 😂 I switched from Java to Go. It’s strange to me that the inheritance of Go. I once had the same idea as you.

File: types/metadataV14.go (hunk 1 above)
Comment (NunoAlexandre) [reply types/metadataV14.go:18]: Thank you!

**Comment by NunoAlexandre** [comment]
@Gkirito thanks again for your help! I am happy to approve and merge once the commented code is deleted.

**Comment by Gkirito** [comment]
> @Gkirito thanks again for your help! I am happy to approve and merge once the commented code is deleted.

OK. I will delete it. Thank you too. I really need this project. Thank you for your contribution to this

**Pull request merged**